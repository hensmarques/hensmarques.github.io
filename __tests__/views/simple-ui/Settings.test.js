import React from 'react';
import { Settings } from '../../../src/views/simple-ui/Settings';

describe('<Settings/>', () => {
  const loadUserSettings = jest.fn();
  const getUserProfile = jest.fn();
  const getKycStatus = jest.fn();
  const kycRequestStatus = jest.fn();

  it('renders a loader spinner and gets user data on mount', () => {
    const settings = {
      status: 'pending',
    };
    const wrapper = shallow(
      <Settings
        settings={settings}
        loadUserSettings={loadUserSettings}
        getUserProfile={getUserProfile}
        getKycStatus={getKycStatus}
        kycRequestStatus={kycRequestStatus}
      />,
    );
    expect(wrapper.find('.loader-container').exists()).toBe(true);
    expect(wrapper.find('Spinner').exists()).toBe(true);
    expect(loadUserSettings).toHaveBeenCalledTimes(1);
    expect(getUserProfile).toHaveBeenCalledTimes(1);
    expect(getKycStatus).toHaveBeenCalledTimes(1);
    expect(kycRequestStatus).toHaveBeenCalledTimes(1);
  });

  it('renders and matches the snapshot', () => {
    const settings = {
      status: 'success',
    };
    const wrapper = shallow(
      <Settings
        settings={settings}
        loadUserSettings={loadUserSettings}
        getUserProfile={getUserProfile}
        getKycStatus={getKycStatus}
        kycRequestStatus={kycRequestStatus}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
