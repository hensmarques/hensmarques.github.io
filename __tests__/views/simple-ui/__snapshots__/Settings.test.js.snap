// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Settings/> renders and matches the snapshot 1`] = `
<Connect(SimpleUiWrapper)
  description="By keeping your account information up to date you allow us to provide the prompt and high quality service you deserve. Your changes will be effective immediately."
  pageClass="settings"
  pageHeader="Update Account Information"
  title="Settings"
>
  <div>
    <Connect(ErrorBoundary)>
      <Connect(AccountInformation)
        translation="SETTINGS"
      />
    </Connect(ErrorBoundary)>
    <Connect(ErrorBoundary)>
      <Connect(SecuritySettings)
        translation="SETTINGS"
      />
    </Connect(ErrorBoundary)>
  </div>
</Connect(SimpleUiWrapper)>
`;
